### PR TITLE
3357 Add wrap_sequence to several transforms

### LINKED ISSUE #3357: Missing `wrap_sequence` in `ToNumpy` and `EnsureType`
These two transforms (also for their dict versions) miss the parameter `wrap_sequence`.

### PR DESCRIPTION
**Pull request by Nic-Ma**
Fixes #3357 .

### Description
This PR added the `wrap_sequence` parameter to several transforms.

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

**Comment by Nic-Ma** [comment]
Hi @rijobro @wyli @yiheng-wang-nv ,

Please note that in this PR I set different default values of `wrap_sequence` for `utils(False)` and `transforms(True)`, following our current pattern in: https://github.com/Project-MONAI/MONAI/blob/dev/monai/utils/type_conversion.py and https://github.com/Project-MONAI/MONAI/blob/dev/monai/transforms/utility/array.py#L356
I think it matches the common usage, we always convert a list of data(like classification labels) to a numpy array or tensor in the transforms.

Thanks.

File: monai/transforms/utility/array.py [hunk 1]
```diff
@@ -377,41 +392,52 @@ def __init__(
...
+        dtype: target data type when converting to numpy array.
+        wrap_sequence: if `False`, then lists will recursively call this function, default to `True`.
+            E.g., if `False`, `[1, 2]` -> `[array(1), array(2)]`, if `True`, then `[1, 2]` -> `array([1, 2])`.
+
     """
 
     backend = [TransformBackends.TORCH, TransformBackends.NUMPY]
 
-    def __init__(self, dtype: DtypeLike = None) -> None:
+    def __init__(self, dtype: DtypeLike = None, wrap_sequence: bool = True) -> None:
```
Comment (yiheng-wang-nv) [inline monai/transforms/utility/array.py:431]: Hi @Nic-Ma , here the default value of `wrap_sequence` has been changed from `False` (used directly in `convert_to_numpy`) to `True`, but I do not see new test case about this class, thus is it not covered?

Comment (Nic-Ma) [reply monai/transforms/utility/array.py:431]: Hi @yiheng-wang-nv ,

The unit tests use `type_test=False` to compare results, so it didn't cover this case, let update to add more tests.

Thanks.

Comment (Nic-Ma) [reply monai/transforms/utility/array.py:431]: Added tests in the latest commit.

Thanks.

**Comment by Nic-Ma** [comment]
Hi @wyli @yiheng-wang-nv ,

Do you have any other concerns about this PR?

Thanks in advance.

**Review by yiheng-wang-nv — APPROVED** [review]
Thanks, looks good on me.

**Pull request merged**